source/agora/common/crypto/ECC.d

```diff
@@ -90,11 +90,43 @@ public struct Scalar
         crypto_core_ed25519_scalar_reduce(this.data[].ptr, param[].ptr);
     }
 
-    /// Expose `toString`
-    public void toString (scope void delegate(const(char)[]) @safe sink)
-        const @safe
+    /// Describe print modes for `toString`
+    public enum PrintMode
     {
-        this.data.toString(sink);
+        /// Print hidden version
+        Obfuscated,
+        /// Print original value
+        Clear,
+    }
+
+    /***************************************************************************
+
+        Print the scalar
+
+        By default, this function prints the hidden version of the scalar.
+        Changing the mode to `Clear` makes it print the original value.
+        This mode is intended for debugging.
+
+        Params:
+          sink = The sink to write the piecemeal string data to
+          mode = The `PrintMode` to use for printing the content.
+                 By default, the hidden value is printed.
+
+    ***************************************************************************/
+
+    public void toString (scope void delegate(const(char)[]) @safe sink,
+                          PrintMode mode = PrintMode.Obfuscated) const @safe
+    {
+        final switch (mode)
+        {
+        case PrintMode.Obfuscated:
+            sink("********");
```

Comment:
```suggestion
            sink("**SEED**");
```

So if we have multiple obfuscated value, we can tell the types apart.

Reply:
Good idea.